cmd/pbstatuser/main.go

```diff
@@ -151,16 +151,24 @@ func checkSourceAvailabilityAWS(ctx context.Context) {
 				Identity:     s.Identity,
 				ResourceType: "Application",
 			}
-			_, err = clients.GetEC2Client(ctx, &s.Authentication, "")
+			ec2Client, err := clients.GetEC2Client(ctx, &s.Authentication, "")
 			if err != nil {
 				sr.Status = kafka.StatusUnavailable
 				sr.Err = err
-				logger.Warn().Err(err).Msg("Could not get aws assumed client")
-				chSend <- sr
+				logger.Warn().Err(err).Msgf("Could not get aws assumed client %s", err)
 			} else {
-				sr.Status = kafka.StatusAvaliable
-				chSend <- sr
+				sr.Status = kafka.StatusAvailable
+			}
+			permissions, err := ec2Client.CheckPermission(ctx, &s.Authentication)
+			if err != nil {
+				sr.Err = err
+				logger.Warn().Err(err).Msgf("Could not check aws permissions %s", err)
+			}
+			sr.MissingPermissions = permissions
+			if len(permissions) != 0 {
+				logger.Warn().Err(err).Msgf("No sufficient permissions: %s", permissions)
```

Comment:
This does not seem correct. CheckPermissions either returns nil result and error, or something in the result. You reported the error above, here I would simply issue a warning with the list. Also can you add a field so we can easily find all records in Kibana, for example:

    logger.Warn().Bool("missing_permission", true).Msgf("No sufficient permissions: %s", permissions)